pkg/services/sqlstore/alert_notification.go

```diff
@@ -192,3 +222,42 @@ func UpdateAlertNotification(cmd *m.UpdateAlertNotificationCommand) error {
 		return nil
 	})
 }
+
+func RecordNotificationJournal(cmd *m.RecordNotificationJournalCommand) error {
+	return inTransaction(func(sess *DBSession) error {
+		journalEntry := &m.NotificationJournal{
+			OrgId:      cmd.OrgId,
+			AlertId:    cmd.AlertId,
+			NotifierId: cmd.NotifierId,
+			SentAt:     cmd.SentAt,
+			Success:    cmd.Success,
+		}
+
+		if _, err := sess.Insert(journalEntry); err != nil {
+			return err
+		}
+
+		return nil
+	})
+}
+
+func GetLatestNotification(cmd *m.GetLatestNotificationQuery) error {
```

Comment:
This handler does not work as expected for me.

It returns and Journal with `&{Id:0 OrgId:0 AlertId:0 NotifierId:0 SentAt:0001-01-01 00:00:00 +0000 UTC Success:false}` I would like this to be tests something like this https://github.com/grafana/grafana/blob/master/pkg/services/sqlstore/alert_notification_test.go#L60

Would be great to test 
* that its possible to record journal events. 
* that we get the correct last notification.
* clean is working as expected. (doesn;'t work currently for me).

Reply:
Let me see if I can reproduce - before the changes I verified in the database that it was returning the correct entry so let me just make sure that it's my most recent commit that broke this.

Reply:
@bergquist I can verify the behavior you're describing but need some input. This only happens on my end when there are no records matching the query that runs which will always be the case before the first notification of the alert. This allows the notification elapsed time to always be above the min interval before the first notification and after that, for me, it returns the correct notification entry. I am still willing to write tests but is this undesirable behavior? If so, what should I do instead?

Reply:
A typed error would be much better since the code will not deal with this scenario in a declarative way. Relaying on the current behavior "works" now but is very fragile since the code/use-case might change in the future.

Reply:
Since this is quite a large feature I will create a new branch for it in the grafana repo and rebase it onto master. It also makes it easier for me to contribute to the PR.

Im you have any unpushed work. just push it to this PR and I will cherry-pick it to the new PR.

https://github.com/grafana/grafana/pull/12145

I will start making some minor changes to the new PR but it would be awesome if you want to contribute some fixes for my last comments + tests.

So im closing this PR in favor of the new one. Please send contributions as PR's to the new branch.